pkg/monitoring/hyperconverged/collectors/operator_collectors.go

```diff
@@ -0,0 +1,83 @@
+package collectors
+
+import (
+	"context"
+
+	"github.com/rhobs/operator-observability-toolkit/pkg/operatormetrics"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hcov1beta1 "github.com/kubevirt/hyperconverged-cluster-operator/api/v1beta1"
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/nodeinfo"
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+)
+
+const (
+	multiArchBootImagesFeatureEnabled  = float64(1)
+	multiArchBootImagesFeatureDisabled = float64(0)
+)
+
+var (
+	multiArchBootImagesStatus = operatormetrics.NewGauge(
+		operatormetrics.MetricOpts{
+			Name: "kubevirt_hco_multi_arch_boot_images_enabled",
+			Help: "indicates if the Multi-Arch Boot Images feature is enabled (1) or not (0)",
+		},
+	)
+)
+
+func getMultiArchBootImagesStatusCollector(cli client.Client, operatorNamespace string) operatormetrics.Collector {
+	return operatormetrics.Collector{
+		Metrics: []operatormetrics.Metric{
+			multiArchBootImagesStatus,
+		},
+		CollectCallback: getMultiArchBootImagesStatusCallback(cli, operatorNamespace),
+	}
+}
+
+func getMultiArchBootImagesStatusCallback(cli client.Client, operatorNamespace string) func() []operatormetrics.CollectorResult {
+	return func() []operatormetrics.CollectorResult {
+		hc := hcov1beta1.HyperConverged{}
+		key := client.ObjectKey{Name: hcov1beta1.HyperConvergedName, Namespace: operatorNamespace}
+		if err := cli.Get(context.TODO(), key, &hc); err != nil {
+			// Don't set the metric if the HyperConverged CR does not exist
+			return []operatormetrics.CollectorResult{}
+		}
+
+		if len(nodeinfo.GetWorkloadsArchitectures()) <= 1 {
+			// Don't set the metric if the cluster is not multi-architecture
+			return []operatormetrics.CollectorResult{}
+		}
+
+		if !ptr.Deref(hc.Spec.EnableCommonBootImageImport, true) &&
+			!enabledCustomDICTsFound(hc.Spec.DataImportCronTemplates) {
+			// Don't set the metric if no common nor custom DataImportCronTemplates found
+			return []operatormetrics.CollectorResult{}
+		}
+
+		// Set the metric based on the FeatureGate value
+		value := multiArchBootImagesFeatureDisabled
```

Comment:
for more clarity I would use similar code as in SetHCOMetricHyperConvergedExists() and SetHCOMetricHyperConvergedNotExists(), see https://github.com/kubevirt/hyperconverged-cluster-operator/blob/main/pkg/monitoring/hyperconverged/metrics/operator_metrics.go#L151

Reply:
This implementation is not working in this case, because we must distinguish between non-set and zero.

The collector is the only option to achieve that.